src/goto-instrument/code_contracts.cpp

```diff
@@ -372,23 +468,39 @@ void code_contractst::add_contract_check(
 
 void code_contractst::operator()()
 {
-  Forall_goto_functions(it, goto_functions)
-    code_contracts(it->second);
-
-  goto_functionst::function_mapt::iterator i_it=
-    goto_functions.function_map.find(INITIALIZE_FUNCTION);
-  assert(i_it!=goto_functions.function_map.end());
+  if(mode_assert_ensures)
+  {
+    Forall_goto_functions(it, goto_functions)
+      assert_ensures_after_calls(it->second);
```

Comment:
Does this share any code with `code_contracts`? If so perhaps it should be a different pass rather than two modes of the same one